rasa/server.py

```diff
@@ -1030,11 +1030,13 @@ async def load_model(request: Request) -> HTTPResponse:
                     {"parameter": "model_server", "in": "body"},
                 )
 
-        app.agent = await _load_agent(
+        tmp_agent = await _load_agent(
             model_path, model_server, remote_storage, endpoints, app.agent.lock_store
         )
+        if tmp_agent.model_directory:
```

Comment:
I check the model in here, I can't make sure this position is the best position to do check. Have any advice?